src/iexamine.cpp

```diff
@@ -2280,6 +2280,164 @@ void iexamine::tree_hickory(player &p, const tripoint &examp)
     none( p, examp );
 }
 
+item_location maple_tree_sap_container() {
+    const item maple_sap = item( "maple_sap", 0 );
+    return g->inv_map_splice( [&]( const item &it ) { 
+        return it.has_valid_capacity_for_liquid( maple_sap, true ) == item::LIQUID_FILL_ERROR::L_ERR_NONE;
```

Comment:
Instead of making `has_valid_capacity_for_liquid` and the enum public, you could use `item::get_remaining_capacity_for_liquid`. Check whether it returns non-zero (which indicates the item can hold the liquid).